### PR TITLE
(CDAP-3627) Merge the HttpContentConsumer and NonTransactionHttpContentConsumer

### PR DESCRIPTION
**Pull request by chtyim**
- In most use cases, transaction is not needed in the onReceive method

### CODE REVIEW

File: cdap-api/src/main/java/co/cask/cdap/api/service/http/HttpContentConsumer.java [hunk 1]
```diff
@@ -71,20 +72,15 @@
    * This method will get invoked when a new chunk of the request body is available to be consumed.
    * It is guaranteed that no concurrent calls to this method will be made.
    * <p>
-   * Also note that each invocation
-   * to this method is executed inside a single transaction, hence access to transactional {@link Dataset Datasets}
-   * is possible through the {@link HttpServiceContext} provided to the
-   * {@link HttpServiceHandler#initialize(HttpServiceContext)} method.
+   * Access to transactional {@link Dataset} should be done through the
```
Comment (anew) [inline cdap-api/src/main/java/co/cask/cdap/api/service/http/HttpContentConsumer.java]: Datasets

**Comment by anew** [comment]
One nitty comment, otherwise LGTM

File: cdap-api/src/main/java/co/cask/cdap/api/service/http/HttpContentConsumer.java (hunk 1 above)
Comment (chtyim) [inline cdap-api/src/main/java/co/cask/cdap/api/service/http/HttpContentConsumer.java]: fixed

**Comment by chtyim** [comment]
Build passed: http://builds.cask.co/browse/CDAP-DUT2913-5

Squashing and merging.

**Pull request merged**